server/app/services/userService.js

```diff
@@ -61,6 +61,85 @@ module.exports.getCharacters = function(region,id,callback){
     });
 };
 
+module.exports.getGuildRank = function (id,region,realm,name,callback){
```

Comment:
This function could have also hit bnet directly, but it will be called more often than `isMember` and `isOwner` are currently called and I didn't want anything tied directly to a page refresh making spurious requests to bnet. As such, rank permissions are evaluated based on WarcraftLFG's view of a guild in MongoDB.